root_VS2012/programs/C#/Frameworks/Infrastructure/Public/Db/DamSqlDbWithMultiShard/DamSqlDbWithMultiShard/MultiShardConfiguration.cs

```diff
@@ -0,0 +1,245 @@
+            {
+                return _shardMapManagerDatabaseName;
+            }
+            set
+            {
+                _shardMapManagerDatabaseName = value;
+            }
+        }
+
+        /// <summary>
+        /// Gets the edition to use for Shards and Shard Map Manager Database if the server is an Azure SQL DB server. 
+        /// If the server is a regular SQL Server then this is ignored.
+        /// </summary>
+        public static string DatabaseEdition
+        {
+            get
+            {
+                return _databaseEdition;
+            }
+        }
+
+        #endregion
+
+        #region Shard map helper methods
+        
+        /// <summary>
+        /// Gets the shard map, if it exists. If it doesn't exist, returns null.
+        /// </summary>
+        /// <returns>shardMap</returns>
+        public static RangeShardMap<int> TryGetShardMap()
+        {
+            if (objShardMapManager == null)
+            {
+                return null;
+            }
+
+            RangeShardMap<int> shardMap;
+            bool mapExists = objShardMapManager.TryGetRangeShardMap(ShardMapName, out shardMap);
+
+            if (!mapExists)
+            {
+                return null;
+            }
+
+            return shardMap;
+        }
+
+        #endregion
+
+        #region connectionstring methods
+
+        /// <summary>
+        /// To get the connection string based on servername,shardmap databasename that are defined in the configuration file
+        /// </summary>
+        /// <returns>Connection strings that are defined in the configuration file</returns>
+        public static string GetConnectionString()
+        {
+            SqlConnectionStringBuilder sbConnStr = new SqlConnectionStringBuilder(connStr);
+            sbConnStr.DataSource = ShardMapManagerServerName;
+            sbConnStr.InitialCatalog = ShardMapManagerDatabaseName;
+            return sbConnStr.ToString();
+        }
+
+        /// <summary>
+        /// To get the connection string based on servername,master databasename that are defined in the configuration file
+        /// </summary>
+        /// <returns>Connection strings that are defined in the configuration file</returns>
+        public static string GetConnectionStringByMasterDatabase()
+        {
+            SqlConnectionStringBuilder sbConnStr = new SqlConnectionStringBuilder(connStr);
+            sbConnStr.DataSource = ShardMapManagerServerName;
+            sbConnStr.InitialCatalog = MasterDatabaseName;
+            return sbConnStr.ToString();
+        }
+
+        /// <summary>
+        /// To open the connection based on specified shard from ShardMap and connectionstring
+        /// </summary>
+        /// <param name="connstring">connstring that is defined in the configuration file</param>
+        /// <returns>Connection open for specific shard</returns>
+        public static SqlConnection GetDataDependentRoutingConnectionString(string connstring)
```

Comment:
@daisukenishino 

>Please change this method name
to "GetDataDependentRoutingConnection"
from "GetDataDependentRoutingConnectionString".

Ok.

>And, I think that this change will also affect the sample.

No this change will not affect the sample.
This will be affected in [damSqlSvr.cs](https://github.com/SymphonyTeleca/OpenTouryoTemplates/blob/damframework-chngs/root_VS2012/programs/C%23/Frameworks/Infrastructure/Business/Business/MyFcBaseLogic.cs#L307)